src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -137,21 +142,22 @@ public void run() {
      * which things are no longer associated with the core device.
      */
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    private Optional<ThingAssociations> getThingsAssociatedWithCoreDevice() {
+    private Optional<Stream<Thing>> getThingsAssociatedWithCoreDevice() {
         RetryUtils.RetryConfig retryConfig = RetryUtils.RetryConfig.builder()
                 .initialRetryInterval(Duration.ofSeconds(30)).maxRetryInterval(Duration.ofMinutes(3))
                 .maxAttempt(3)
                 .retryableExceptions(Arrays.asList(ThrottlingException.class, InternalServerException.class)).build();
 
         try {
-            Stream<Thing> cloudThings = RetryUtils.runWithRetry(
+            Stream<List<AssociatedClientDevice>> cloudAssociatedDevices = RetryUtils.runWithRetry(
                     retryConfig, iotAuthClient::getThingsAssociatedWithCoreDevice,
                             "get-things-associated-with-core-device", logger);
 
-            Stream<Thing> localThings = thingRegistry.getAllThings();
-            ThingAssociations associations =  ThingAssociations.create(cloudThings);
-            associations.setLocalThings(localThings);
-            return Optional.of(associations);
+            Stream<Thing> cloudThings = cloudAssociatedDevices.flatMap(List::stream)
```

Comment:
nit: i think flatMap would fit better in `IotAuthClient#getThingsAssociatedWithCoreDevice`, lets us abstract paging details and easier to reason about the code.